drivers/mge-hid.c

```diff
@@ -1151,6 +1151,119 @@ static info_lkp_t eaton_input_bypass_mode_off_info[] = {
 	{ 0, NULL, NULL, NULL }
 };
 
+/* Function to start or stop ECO(HE) Mode automatically instead of manually starting/stoping Bypass and then ECO(HE) Mode */
+static const char *eaton_input_eco_mode_auto_on_off_fun(double value)
+{
+	const char *bypass_switch_str = NULL;
+	const char *eco_switchable_str = NULL;
+
+	if ((int)value == 1) {
+		/* Check if input.bypass.switch.on is disabled and set it to 'on' */
+		bypass_switch_str = dstate_getinfo("input.bypass.switch.on");
+		if (!strcmp(bypass_switch_str, "disabled")) {
+			bypass_switch_str = eaton_input_bypass_check_range(value);
+		} else {
+			upsdebugx(1, "%s: Bypass switch on state is: %s , must be disabled before switching on", __func__, bypass_switch_str);
+			return NULL;
+		}
+
+		/* Check if input.eco.switchable is normal and set it to 'ECO' */
+		eco_switchable_str = dstate_getinfo("input.eco.switchable");
+		if (!strcmp(eco_switchable_str, "normal")) {
+			eco_switchable_str = eaton_input_eco_mode_check_range(value);
+		} else {
+			upsdebugx(1, "%s: ECO switch state is: %s , must be normal before switching to ECO", __func__, eco_switchable_str);
+			return NULL;
+		}
+
+		upsdebugx(1, "%s: ECO Mode was enabled after switching to Bypass Mode", __func__);
+        return "on";
+
+	} else {
```

Comment:
@masterwishx : sanity-check: I think this method should accept `0` as input here explicitly, and error out verbosely for unsupported inputs (just in case, "2 / ESS" was somewhere else, right?)